src/lib/model/CertificatesModel.ts

```diff
@@ -0,0 +1,288 @@
+/**************************************************************************
+ *  (C) Copyright ModusBox Inc. 2020 - All rights reserved.               *
+ *                                                                        *
+ *  This file is made available under the terms of the license agreement  *
+ *  specified in the corresponding source code repository.                *
+ *                                                                        *
+ *  ORIGINAL AUTHOR:                                                      *
+ *       Murthy Kakarlamudi - murthy@modusbox.com                   *
+ **************************************************************************/
+
+import Vault from '../vault';
+import { DFSPCertificateModel, HTTPResponseError, HubCertificateModel } from '@pm4ml/mcm-client';
+
+import assert from 'assert';
+import forge from 'node-forge';
+import ConnectorManager from './ConnectorManager';
+import { Logger } from '@mojaloop/sdk-standard-components';
+import { Knex } from 'knex';
+
+export interface CertificatesModelOpts {
+  logger: Logger.Logger;
+  vault: Vault;
+  db: Knex;
+  dfspId: string;
+  mcmServerEndpoint: string;
+}
+
+class CertificatesModel {
+  private _logger: Logger.Logger;
+  private _vault: Vault;
+  private _db: Knex;
+  private _outboundCert?: string;
+  private _connectorManager: ConnectorManager;
+  private _mcmClientDFSPCertModel: DFSPCertificateModel;
+  private _certificateModel: HubCertificateModel;
+
+  constructor(opts: CertificatesModelOpts) {
+    this._logger = opts.logger;
+    this._vault = opts.vault;
+    this._db = opts.db;
+
+    this._mcmClientDFSPCertModel = new DFSPCertificateModel({
+      dfspId: opts.dfspId,
+      logger: opts.logger,
+      hubEndpoint: opts.mcmServerEndpoint,
+    });
+
+    this._certificateModel = new HubCertificateModel({
+      dfspId: opts.dfspId,
+      logger: opts.logger,
+      hubEndpoint: opts.mcmServerEndpoint,
+    });
+
+    this._connectorManager = new ConnectorManager(opts);
+  }
+
+  async uploadClientCSR(body) {
+    return this._mcmClientDFSPCertModel.uploadCSR({
+      csr: body,
+    });
+  }
+
+  async createCSR(keyBits, csrParameters: any = {}) {
+    const keys = forge.pki.rsa.generateKeyPair(keyBits);
+    const csr = forge.pki.createCertificationRequest();
+    csr.publicKey = keys.publicKey;
+    if (csrParameters?.subject) {
+      csr.setSubject(Object.entries(csrParameters.subject).map(([shortName, value]) => ({ shortName, value })));
+    }
+    if (csrParameters?.extensions?.subjectAltName) {
+      const DNS_TYPE = 2;
+      const IP_TYPE = 7;
```

Comment:
I either recommend using well-defined enums to describe the meaning of the 7 values for `IP_TYPE` or adding a comment to the assignment.